xlators/cluster/afr/src/afr-common.c

```diff
@@ -4161,11 +4122,6 @@ __afr_fd_ctx_set(xlator_t *this, fd_t *fd)
 
```

Comment:
nit: the line
```
VALIDATE_OR_GOTO(fd, out);
```

Is not needed, as this whole call is (now clearly) under fd->lock, so fd cannot be NULL.

Reply:
I think this is okay. In fact right thing to do. There is no way we are stopping anyone
using this function without lock. I mean developer can use this function without lock
and could introduce a bug. 
So, if we can not control from where we can land up in a function, it is better to control
what we are getting in a function.

Reply:
This function is static, so only used in this file.
We can add documentation (and we should, not just use the _notation) to mark that it should be called under lock.

Most likely the compiler optimizes this redundant call anyway (easy to check).

We can't control functions we export. We can control static functions and we can and should expect developers to carefully code.

Reply:
I still think that between the trade off of saving time for one instruction on local system and a rare chance of crash, we should choose to avoid crash. We should not depend on developers wisdom to avoid crash in production.
Having said that, I would make this change here.